### PR TITLE
Add delete authenticator mutation

### PR DESCRIPTION
**Pull request by kiootic**
ref #312

### CODE REVIEW

File: pkg/lib/interaction/nodes/do_remove_authenticator.go [hunk 1]
```diff
@@ -29,11 +31,63 @@ func (n *NodeDoRemoveAuthenticator) Prepare(ctx *interaction.Context, graph *int
...
+		switch n.Authenticator.Kind {
+		case authenticator.KindPrimary:
+			// Ensure all identities have matching primary authenticator.
+			is, err := ctx.Identities.ListByUser(userID)
 			if err != nil {
 				return err
 			}
+
+			aTypes := map[authn.AuthenticatorType]struct{}{}
+			for _, a := range filterAuthenticators(as, authenticator.KeepKind(authenticator.KindPrimary)) {
```
Comment (louischan-oursky) [inline pkg/lib/interaction/nodes/do_remove_authenticator.go:53]: Why aren't we using `KeepPrimaryAuthenticatorOfIdentity`?

Comment (kiootic) [reply pkg/lib/interaction/nodes/do_remove_authenticator.go:53]: Didn't notice that. Let me change to use it.

**Comment by louischan-oursky** [comment]
Given this query

```graphql
query {
  users {
    edges {
      node {
        id
        createdAt
        authenticators {
          edges {
            node {
              id
              type
              kind
              isDefault
            }
          }
        }
      }
    }
  }
}

```

and result 

```json
{
  "data": {
    "users": {
      "edges": [
        {
          "node": {
            "authenticators": {
              "edges": [
                {
                  "node": {
                    "id": "QXV0aGVudGljYXRvcjpvb2Jfb3RwfDdlZmMzZTBhLTQ3YTctNGZlMi1iYTViLWRkMWUzZmFjMjYwZg==",
                    "isDefault": true,
                    "kind": "SECONDARY",
                    "type": "OOB_OTP"
                  }
                },
                {
                  "node": {
                    "id": "QXV0aGVudGljYXRvcjpwYXNzd29yZHwxYWIyM2JiMy02Njg2LTQxNjYtOTkyMy01NmFhZmZlMDM2ZTE=",
                    "isDefault": false,
                    "kind": "PRIMARY",
                    "type": "PASSWORD"
                  }
                }
              ]
            },
            "createdAt": "2020-09-15T11:17:08.009533Z",
            "id": "VXNlcjo0ODRhODFkYS0zZGEyLTRkODEtOWNiMS0yYTkzNTdlZmJkM2I="
          }
        }
      ]
    }
  }
}
```

I can remove my primary password. But I guess this should not be possible.

P.S. My login ID is an email and my 2FA OOB OTP is the same email.

**Comment by kiootic** [comment]
It turns out `KeepPrimaryAuthenticatorOfIdentity` does not check the authenticator kind to be primary. I'll fix it as well.

**Pull request merged**